src/neuroconv/tools/nwb_helpers/_models/_base_dataset_models.py

```diff
@@ -0,0 +1,180 @@
+"""Base Pydantic models for DatasetInfo and DatasetConfiguration."""
+import math
+from abc import ABC, abstractmethod
+from typing import Any, Dict, Literal, Tuple, Union
+
+import h5py
+import numcodecs
+import numpy as np
+from pydantic import BaseModel, Field, root_validator, validator
+
+
+class DatasetInfo(BaseModel):
```

Comment:
Can you add a small docstring here clarifying how is this different from `DatasetConfiguration(BaseModel,`. If I remember well our in-person discussion you want this object to correspond to the inmutable data so let's keep the docstring as a reminder on why these concepts were separated as a clarification of intent for future developers (versions of us most likely).

Reply:
Done

Reply:
Thanks.